common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java

```diff
@@ -0,0 +1,73 @@
+package org.sunbird.common.request;
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.responsecode.ResponseCode;
+
+public class UserProfileRequestValidator extends BaseRequestValidator {
+
+  @SuppressWarnings("unchecked")
+  public void validateProfileVisibility(Request request) {
+    if (request.getRequest().get(JsonKey.USER_ID) == null
```

Comment:
BaseRequestValidator.validateParam can be used with mandatoryParamsMissing error as there is no check being done on username.